scalafmt-tests/src/test/resources/default/String.stat

```diff
@@ -360,3 +360,90 @@ object X {
       .f("  sldkfjdslkf sdlkfj dslkfjds flkdsjf lsdkjf dslkfjds sdklfjsd sdlkjfdskj")
   }
 }
+<<< #3079 !newlines.inInterpolation
+maxColumn = 50
+newlines.inInterpolation = false
+===
+object a {
+  s"foo1 ${bar} foo2 ${baz(bar)} foo3 ${qux(baz, bar)} foo4"
+  s"""
+    |foo1 ${bar} foo2 ${baz(bar)} foo3 ${qux(baz, bar)} foo4
+    |""".stripMargin
+}
+>>>
+object a {
+  s"foo1 ${bar} foo2 ${baz(bar)} foo3 ${qux(baz, bar)} foo4"
+  s"""
+    |foo1 ${bar} foo2 ${baz(bar)} foo3 ${qux(baz, bar)} foo4
+    |""".stripMargin
+}
+<<< #3079 newlines.inInterpolation
+maxColumn = 50
+newlines.inInterpolation = true
+===
+object a {
+  s"foo1 ${bar} foo2 ${baz(bar)} foo3 ${qux(baz, bar)} foo4"
+  s"""
+    |foo1 ${bar} foo2 ${baz(bar)} foo3 ${qux(baz, bar)} foo4
+    |""".stripMargin
+}
+>>>
+object a {
+  s"foo1 ${bar} foo2 ${baz(bar)} foo3 ${qux(baz, bar)} foo4"
+  s"""
+    |foo1 ${bar} foo2 ${baz(bar)} foo3 ${qux(
+      baz,
+      bar)} foo4
+    |""".stripMargin
+}
+<<< #3079 newlines.inInterpolation, source=keep
+maxColumn = 50
+newlines.source = keep
+newlines.inInterpolation = true
+includeCurlyBraceInSelectChains = true
+===
+object a {
+  s"foo1 ${
+  bar} foo2 ${baz(bar)} foo3 ${qux(baz, bar)} foo4"
+  s"""
+    |foo1 ${bar} foo2 ${baz(bar)} foo3 ${qux(baz, bar)} foo4
+    |""".stripMargin
+}
+>>>
+object a {
+  s"foo1 ${
+      bar
+    } foo2 ${baz(bar)} foo3 ${qux(baz, bar)} foo4"
+  s"""
+    |foo1 ${bar} foo2 ${baz(bar)} foo3 ${qux(
+      baz,
+      bar)} foo4
+    |""".stripMargin
+}
+<<< #3079 newlines.inInterpolation, source=keep, align
+maxColumn = 50
+newlines.source = keep
+newlines.inInterpolation = true
+align.inInterpolation = true
+includeCurlyBraceInSelectChains = true
+===
+object a {
+  s"foo1 ${
+  bar} foo2 ${baz(bar)} foo3 ${
+  qux(baz, bar)} foo4"
+  s"""
+    |foo1 ${bar} foo2 ${baz(bar)} foo3 ${qux(baz, bar)} foo4
+    |""".stripMargin
+}
+>>>
+object a {
+  s"foo1 ${
+            bar
+          } foo2 ${baz(bar)} foo3 ${
+                                     qux(baz, bar)
+                                   } foo4"
+  s"""
+    |foo1 ${bar} foo2 ${baz(bar)} foo3 ${qux(baz,
+                                             bar)} foo4
```

Comment:
Shouldn't there be an additional break here? `foo4` seems to reach maxColumn (51) if I calculated correctly.

Reply:
ah... let me check

Reply:
you were right. added another option, oneline, which is least likely to overflow, and mentioned in the docs the caveat about the others. please take another look.